### PR TITLE
Refactor routes

### PR DESCRIPTION
**Pull request by beligante**

### CODE REVIEW

File: utils/url-params.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
+const _ = require('lodash')
+const TAGS = [
```
Comment (gabriel-laet) [inline utils/url-params.js]: you should retrieve these from API somehow. make a single request & put tags list in memory

File: __tests__/utils/url-params.test.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      city: 'city',
+      filters: {
+        price: {min: 10000, max: 1000000000},
+        rooms: {min: 3, max: 3}
+      },
+      tags: []
+    }
+    expect(result).toEqual(expected)
+  })
+})
```
Comment (gabriel-laet) [inline __tests__/utils/url-params.test.js]: make sure you add tests for existing URLs that shouldn't break (like :street/id-)

File: utils/url-params.js (hunk 1 above)
Comment (gabriel-laet) [reply utils/url-params.js]: I think you can release like that but we should create a new PR that solves the problem of having to whitelist all tags in multiple places

Comment (beligante) [reply utils/url-params.js]: My idea is to fetch that throughout GraphQL, I just made that way because it's less complex to stat the refactoring

**Comment by beligante** [comment]
> I'd separate removing `messages` in another PR

Ok! I will undo the removing! Make sense

File: __tests__/utils/url-params.test.js (hunk 2 above)
Comment (beligante) [reply __tests__/utils/url-params.test.js]: This is just for the parser to transform the paths into tags, features or nighborhoods. I believe that I will create another router in express to handle streets with id

**Pull request merged**